template/src/navigation/index.tsx

```diff
@@ -14,6 +17,22 @@ function Navigation() {
 
   useKillswitch();
 
+  useEffect(() => {
+    // Refetch queries on app focus
+    const subscription = AppState.addEventListener('change', (appState) => {
+      focusManager.setFocused(appState === 'active');
+    });
+
+    return () => subscription.remove();
+  }, []);
+
+  useEffect(() => {
+    // Refetch queries on reconnect
+    return NetInfo.addEventListener((state) => {
+      onlineManager.setOnline(!!state.isConnected);
+    });
+  }, []);
```

Comment:
Can we simplify this in one useEffect? And also maybe we could extract this in its own custom hook (i.e. useRefetchQueries)

Reply:
I made it two separate hooks since it doesn't do the same thing and maybe sometimes you don't want one or the other so it's gonna be easier to remove only one of them.